### ISSUE TITLE
Help us translate these roadmaps

### ISSUE THREAD

**Issue by kamranahmedse** [kamranahmedse/developer-roadmap#669]
Please follow the below steps if you would like to help translate these.

* Fork this repository.
* Create a translation file at `translations/[language].md` e.g. `translations/urdu.md`
* Write down the translated text in the form of markdown table for all of the boxes in the roadmap e.g.

  | **English**            | **Translation**              |
  |------------------------|------------------------------|
  | How does Internet Work | انٹرنیٹ کیسے کام کرتا ہے     |
  | DNS and how it works?  | اور یہ کیسے کام کرتا ہے؟ DNS |

* Create a pull request with your changes.

**Comment by hszhakka** [comment]
I can help you translate it into Chinese.

**Comment by hszhakka** [comment]
How would you exploit these translations? I think directly translate the `text` values in the JSON files would be a better method. For example, copy `backend-map.json` into `backend-map-cn.json`, then directly replace `text` values with relative Chinese text.

**Comment by kamranahmedse** [comment]
@hszhakka contributing and maintaining the translations using JSON files isn't going to be easier. We will parse the translation files and dynamically generate the JSON for those using a script.

This will make the later updates to the translations also easier.

**Comment by hszhakka** [comment]
> @hszhakka contributing and maintaining the translations using JSON files isn't going to be easier. We will parse the translation files and dynamically generate the JSON for those using a script.
> 
> This will make the later updates to the translations also easier.

OK. I'll do the translation quickly.

**Comment by yufieko** [comment]
I can help with Indonesian translation. Before that, I have a question,

(1) 
![image](https://user-images.githubusercontent.com/7820512/77828779-262c9600-7150-11ea-9f0d-1efc17dd5258.png) 

(2)
![image](https://user-images.githubusercontent.com/7820512/77828866-aa7f1900-7150-11ea-9405-9025608a5f67.png) 

(3)
![image](https://user-images.githubusercontent.com/7820512/77828891-ca164180-7150-11ea-8313-45da00115dad.png) 

with the table format above, will the sentences be separated by new rows or in one row?
especially in pictures number one and two.

**Comment by kamranahmedse** [comment]
@yufieko each sentence should be a separate row, irrespective of how they are organized in the UI. In the cases that you have mentioned, here are different rows:

* Personal Recommendation / Opinion
* Alternative Option - Pick this or purple
* Order in roadmap not strict (Learn anytime)
* I wouldn't recommend
* CSS Preprocessors
* With how the modern frameworks work, there has been more push towards...
* Bootstrap
* Materialize CSS
* Bulma
* CSS first frameworks which don't come with JavaScript components by default

**Comment by thezaff** [comment]
> Please follow the below steps if you would like to help translate these.
> 
> * Fork this repository.
> * Create a translation file at `translations/[language].md` e.g. `translations/urdu.md`
> * Translate all the boxes in the roadmap in below format e.g.
>   
>   
>   
>   **English**
>   **Translation**
>   
>   
>   
>   
>   How does Internet Work
>   انٹرنیٹ کیسے کام کرتا ہے
>   
>   
>   DNS and how it works?
>   اور یہ کیسے کام کرتا ہے؟ DNS
> * Create a pull request with your changes.

Hi, @kamranahmedse 

I think it would be pretty easier to translate, if we had a template file (table). One column with all box names in English, and another empty column for translated box names.

**Comment by hszhakka** [comment]
Yes! Maybe you could write a script to export a template file. @kamranahmedse

**Comment by Houssk** [comment]
Do you need some help for french ?

**Comment by munizigor** [comment]
I can help with Portuguese-BR.

**Comment by marefati110** [comment]
i can help you translate it to persian
its good to have a general structure for translation

**Comment by kamranahmedse** [comment]
Actually, I went through the JSON exports and it wouldn't be easier to created translated roadmaps from the markdown translations. I have updated the contribution docs in the issue. Please have a look, you can just update the project files and update the roadmaps in Balsamiq. @hszhakka @thezaff @yufieko 

@munizigor @marefati110 please go ahead with the translation :)

**Comment by sizplay** [comment]
I can translate it to Korean
It will be so interesting, please let me know

**Comment by clRcoDE** [comment]
i translated the front-end part to Farsi/Persian
the pull request created and it's available for merge now
that was the first time i was contributing to a github project and this make me kinda ... happy 🥳😬
lastly i wanna mention something about your json/balsamiq approach for this : its awesome 😗👌
seriously it is easy and clean btw
#happy_coding_in_quarantine

**Comment by ScarboroughCoral** [comment]
I can help you translate the front-end roadmap into Chinese.

**Comment by MarioAGR** [comment]
I would like to help with Spanish translation, throughout tomorrow you'll have it.

**Comment by kamranahmedse** [comment]
Hey guys, thank you for offering to help. Please don't comment here. If you would like to translate and the translation doesn't exist already please go ahead with the translation.

**Comment by ccloli** [comment]
@hszhakka I've already made a Simplified Chinese translation fork (https://github.com/ccloli/developer-roadmap-zh-CN), maybe you can review it, HTH.

**Comment by hszhakka** [comment]
> @hszhakka I've already made a Simplified Chinese translation fork (https://github.com/ccloli/developer-roadmap-zh-CN), maybe you can review it, HTH.

Thanks. @ScarboroughCoral and I already had translated the front-end and back-end roadmap into Chinese and sent PRs, so maybe you can translate the DevOps roadmap.

**Comment by lalizita** [comment]
I can help translate ir to portuguese-BR

**Comment by H1rkul2** [comment]
I can help with arabic

**Comment by humanier** [comment]
I can do Ukrainian and Russian no prob

**Comment by kamranahmedse** [comment]
@lalizita thank you we already have that translated.

@H1rkul2 @humanier thank you, please go ahead with the translation!

**Comment by Shohan494** [comment]
I am interested to convert this in Bengali

**Comment by armurad** [comment]
Hi There, which one you are talking about?


Regards,
Abdur

Sent on the go with Vodafone
________________________________

**Comment by ranasaani** [comment]
Hi Kamran,
I can help with Urdu translation.

**Comment by kamranahmedse** [comment]
Thanks @ranasaani – feel free to do it :)

**Comment by Upinder123** [comment]
hi Kamran i can help with hindi translations

**Comment by FabianoLothor** [comment]
> * Export the JSON file Project > Export > Mockup to JSON and put it at `translations/[language]/src/[name].json` e.g. `translations/portugese/src/backend-map.json`
> * Export the image Project > Export > Current Mockup to PNG and put it in the `trans/[language]/img/[name].png` e.g. `trans/portugese/img/backend-map.png`

You should correct your post: `portuguese` instead of `portugese`.

**Comment by H1rkul2** [comment]
taking care of the devops arabic map this weekend
